### PR TITLE
Increase default part-size to 64MB and provide a way to over-ride the default part size

### PR DESCRIPTION
**Pull request by krishnasrinivas**

### CODE REVIEW

**Comment by ebozduman** [comment]
@krishnasrinivas ,
Could you look into the travis failure?

**Comment by ebozduman** [comment]
Adding a functional test could be a good idea.

**Comment by ebozduman** [comment]
`README.md` and `README_zh_CN.md` files need to be updated and missing parameters need to be added into them.

File: src/main/minio.js [hunk 1]
```diff
@@ -244,20 +247,22 @@ export class Client {
...
     }
     if (size > this.maxObjectSize) {
       throw new TypeError(`size should not be more than ${this.maxObjectSize}`)
     }
-    var partSize = Math.ceil(size/10000)
-    partSize = Math.ceil(partSize/this.minimumPartSize) * this.minimumPartSize
-    return partSize
+    var partSize = this.partSize
+    while (true) {
+      if ((partSize * 10000) > size) {
```
Comment (kannappanr) [inline src/main/minio.js]: Why was the old code removed? It looks good to me. Even if the user provides a `partsize`, will we override that value here?

**Comment by krishnasrinivas** [comment]
@ebozduman @kannappanr can you check now

File: src/main/minio.js [hunk 2]
```diff
@@ -244,20 +247,22 @@ export class Client {
...
+	return partSize
+      }
+      // Try part sizes as 64, 90, 106 etc.
```
[the remark this answers is not in the document]
Comment (ebozduman) [reply src/main/minio.js]: @krishnasrinivas,
I see the increment issue is fixed as `16 --> 16*1024*1024`.

However the incremented possible part size values `90` and `106` in the comment are still there. `90` and `106` are not multiples of 16.
The comment is supposed to be `Try part sizes  64MB,`**`80MB, 96MB, 112MB,`**`... etc.`

File: src/main/minio.js [hunk 3]
```diff
@@ -244,20 +248,26 @@ export class Client {
...
+    if (this.overRidePartSize) {
+      return this.partSize
+    }
+    var partSize = this.partSize
+    while (true) {
+      // If partSize is big enough to accomodate the object size, then use it.
+      if ((partSize * 10000) > size) {
+	return partSize
+      }
+      // Try part sizes as 64MB, 90MB, 106MB etc.
```
Comment (kannappanr) [inline src/main/minio.js]: `// Try part sizes as 64MB, 90MB, 106MB etc.` should be `// Try part sizes as 64MB, 80MB, 96MB etc.`

**Comment by ebozduman** [comment]
@krishnasrinivas,
How about the missing changes in documentation files, `docs/zh_CN/API.md`, `README.md` and `README_zh_CN.md`?

**Comment by harshavardhana** [comment]
ping @krishnasrinivas can you make the changes requested by @ebozduman ?

File: src/main/minio.js (hunk 1 above)
Comment (krishnasrinivas) [reply src/main/minio.js]: > Even if the user provides a partsize, will we override that value here?

No, if user overrides, then that value is used for the partSize

**Comment by krishnasrinivas** [comment]
@kannappanr @ebozduman can you review the PR again? your comments have been addressed.

File: src/main/minio.js (hunk 1 above)
Comment (kannappanr) [reply src/main/minio.js]: ok

**Pull request merged**